File: _wpf/wpf-business-programmer/3-wpf-business-programmer-m3-exercise-files/Samples/CSharp/WPFTemplateApp/WPFTemplateApp/DataClasses/Product.cs

namespace WPFTemplateApp
{
  public class Product : ObjectBase
  {
    #region Private Variables
    private int _ProductId = 0;
    private string _ProductName = string.Empty;
    private decimal _Price = 0;
    private bool _IsActive = true;
    #endregion

    #region Public Properties
    public int ProductId
    {
      get { return _ProductId; }
      set
      {
        if (_ProductId != value)
        {
          _ProductId = value;
          RaisePropertyChanged("ProductId");
        }
      }
    }

    public string ProductName
    {
      get { return _ProductName; }
      set
      {
        if (_ProductName != value)
        {
          _ProductName = value;
          RaisePropertyChanged("ProductName");
        }
      }
    }

    public decimal Price
    {
      get { return _Price; }
      set
      {
        if (_Price != value)
        {
          _Price = value;
          RaisePropertyChanged("Price");
        }
      }
    }

    public bool IsActive
    {
      get { return _IsActive; }
      set
      {
        if (_IsActive != value)
        {
          _IsActive = value;
          RaisePropertyChanged("IsActive");
        }
      }
    }
    #endregion
  }
}

File: _wpf/wpf-business-programmer/3-wpf-business-programmer-m3-exercise-files/Samples-2012/VB/WPFTemplateApp/WPFTemplateApp/DataClasses/Product.vb
Public Class Product
  Inherits ObjectBase
#Region "Private Variables"
  Private _ProductId As Integer = 0
  Private _ProductName As String = String.Empty
  Private _Price As Decimal = 0
  Private _IsActive As Boolean = True
#End Region

#Region "Public Properties"
  Public Property ProductId() As Integer
    Get
      Return _ProductId
    End Get
    Set(ByVal value As Integer)
      If _ProductId <> value Then
        _ProductId = value
        RaisePropertyChanged("ProductId")
      End If
    End Set
  End Property

  Public Property ProductName() As String
    Get
      Return _ProductName
    End Get
    Set(ByVal value As String)
      If _ProductName <> value Then
        _ProductName = value
        RaisePropertyChanged("ProductName")
      End If
    End Set
  End Property

  Public Property Price() As Decimal
    Get
      Return _Price
    End Get
    Set(ByVal value As Decimal)
      If _Price <> value Then
        _Price = value
        RaisePropertyChanged("Price")
      End If
    End Set
  End Property

  Public Property IsActive() As Boolean
    Get
      Return _IsActive
    End Get
    Set(ByVal value As Boolean)
      If _IsActive <> value Then
        _IsActive = value
        RaisePropertyChanged("IsActive")
      End If
    End Set
  End Property
#End Region
End Class